### PR TITLE
[MLIR] Vector dialect: Address post-merge review comments on #111541

### PR DESCRIPTION
**Pull request by bjacob**

### CODE REVIEW

**Comment by bjacob** [comment]
@banach-space , see responses on #111541 for the things not changed here, such as the `shape_cast`.

File: mlir/test/Dialect/Vector/vector-contiguous-extract-strided-slice-to-extract.mlir [hunk 1]
```diff
@@ -1,34 +1,25 @@
...
   %1 = vector.extract_strided_slice %arg0 {offsets = [0, 0, 0, 0, 0, 0], sizes = [1, 1, 1, 1, 1, 4], strides = [1, 1, 1, 1, 1, 1]} : vector<8x1x2x1x1x4xi32> to vector<1x1x1x1x1x4xi32>
   %2 = vector.shape_cast %1 : vector<1x1x1x1x1x4xi32> to vector<4xi32>
   return %2 : vector<4xi32>
 }
 
-// CHECK-LABEL: @extract_strided_slice_to_extract_i32_non_contiguous_1
+// CHECK-LABEL: @non_full_size
 // CHECK:        vector.extract_strided_slice
-func.func @extract_strided_slice_to_extract_i32_non_contiguous_1(%arg0 : vector<8x1x2x1x1x4xi32>) -> vector<2xi32> {
+func.func @non_full_size(%arg0 : vector<8x1x2x1x1x4xi32>) -> vector<2xi32> {
```
Comment (banach-space) [inline mlir/test/Dialect/Vector/vector-contiguous-extract-strided-slice-to-extract.mlir:14]: To differentiate from the one below.

```suggestion
func.func @non_full_trailing_size(%arg0 : vector<8x1x2x1x1x4xi32>) -> vector<2xi32> {
```

File: mlir/test/Dialect/Vector/vector-contiguous-extract-strided-slice-to-extract.mlir [hunk 2]
```diff
@@ -1,34 +1,25 @@
...
   %1 = vector.extract_strided_slice %arg0 {offsets = [0, 0, 0, 0, 0, 0], sizes = [1, 1, 1, 1, 1, 2], strides = [1, 1, 1, 1, 1, 1]} : vector<8x1x2x1x1x4xi32> to vector<1x1x1x1x1x2xi32>
   %2 = vector.shape_cast %1 : vector<1x1x1x1x1x2xi32> to vector<2xi32>
   return %2 : vector<2xi32>
 }
 
-// CHECK-LABEL: @extract_strided_slice_to_extract_i32_non_contiguous_2
+// CHECK-LABEL: @non_contiguous
 // CHECK:        vector.extract_strided_slice
-func.func @extract_strided_slice_to_extract_i32_non_contiguous_2(%arg0 : vector<8x1x2x1x1x4xi32>) -> vector<2xi32> {
+func.func @non_contiguous(%arg0 : vector<8x1x2x1x1x4xi32>) -> vector<2xi32> {
```
Comment (banach-space) [inline mlir/test/Dialect/Vector/vector-contiguous-extract-strided-slice-to-extract.mlir:22]: To differentiate from the one above
```suggestion
func.func @non_full_inner_size(%arg0 : vector<8x1x2x1x1x4xi32>) -> vector<2xi32> {
```

**Review by banach-space — COMMENTED** [review]
That's true speed of light Benoit, thank you!

I've left two small suggestions here and under #111541 :
* use `CHECK-NEXT`, and 
* skip checking strides altogether). 

LGTM otherwise!

Apologies for being a bit pedantic, especially when it comes to tests. That's after spending quite some quality time with them 😅 These updates are much appreciated 🙏🏻

**Comment by bjacob** [comment]
Regarding the unit strides: Actually I was wrong earlier, as the pattern will match as soon as there is any full-size slice dimension.  So I went ahead and added a strided testcase... but then it failed to validate, the validator requiring unit strides. And indeed, this is said in the doc: https://mlir.llvm.org/docs/Dialects/Vector/#vectorextract_strided_slice-vectorextractstridedsliceop :
> At the moment strides must contain only 1s.
However, it sounds possible that strides might become flexible in the future, so to limit the risk of bugs popping up then, I'd like to keep the check, even if it can't be exercised now. That's also in line with other patterns in this file.

**Comment by banach-space** [comment]
> Regarding the unit strides: Actually I was wrong earlier, as the pattern will match as soon as there is any full-size slice dimension. So I went ahead and added a strided testcase... but then it failed to validate, the validator requiring unit strides. And indeed, this is said in the doc: https://mlir.llvm.org/docs/Dialects/Vector/#vectorextract_strided_slice-vectorextractstridedsliceop :
> 
> > At the moment strides must contain only 1s.
> > However, it sounds possible that strides might become flexible in the future, so to limit the risk of bugs popping up then, I'd like to keep the check, even if it can't be exercised now. That's also in line with other patterns in this file.

Yeah, so much for "strided" in `extract_strided_slice` 😂 Perhaps add a TODO somewhere ("Add a test for non-unit strides once these are supported")?

**Review by banach-space — APPROVED** [review]
Thanks!

**Pull request merged**